### PR TITLE
Add Zoho as an OAuth provider

### PR DESCRIPTION
**Pull request by cxumol**
Hello

This PR intends to add Zoho as an OAuth provider, as described in https://www.zoho.com/accounts/protocol/oauth/sign-in-using-zoho.html.

I have run `make test`, `make lint`, and `npm run build` according to CONTRIBUTING.md, and all tests and lint checks passed.

Up to the commit https://github.com/pocketbase/pocketbase/commit/351e3b83390fd2763cda1541ab919800bdce153c, my changes were based on the `master` branch. I built the binary from this commit and successfully signed in with the Zoho OAuth provider on this build.

Commits after https://github.com/pocketbase/pocketbase/commit/351e3b83390fd2763cda1541ab919800bdce153c  are trying to align with the `develop` branch (i.e., align with v0.23.0-rc changes). So, this commit, https://github.com/pocketbase/pocketbase/commit/f8c7ecf0f855b5358a498e3c050ff634f0df9cdf also passed `make test` and `make lint`. However, I couldn't validate it in the real world, like what I did with https://github.com/pocketbase/pocketbase/commit/351e3b83390fd2763cda1541ab919800bdce153c, because the authentication settings are missing from the Admin UI on the `develop` branch.

I read https://github.com/pocketbase/pocketbase/pull/4999#pullrequestreview-2081840100 and understand the `auth` package might have further changes. Please let me know when it's stabilized so that I can update this PR.

```HTML
<!DOCTYPE html><html lang="en"><head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>PB-Zoho-Auth</title>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/pocketbase/0.21.5/pocketbase.umd.min.js"></script>
</head>
<body>

  <button onclick="doAuth()">login</button>
  <pre id="response"></pre>
  <hr/>
  <pre id="authData"></pre>

  <script>
  var pb = new PocketBase('http://127.0.0.1:8090');
  function doAuth(){
    pb.collection('users').authWithOAuth2({ provider: 'zoho' }).then((resp)=>{
      document.getElementById('response').innerText=JSON.stringify(resp, null, 2);
      document.getElementById('authData').innerText=JSON.stringify(pb.authStore.model, null, 2);
    }).catch( (err)=>{
      console.log("Failure:", err, err.originalError);
      document.getElementById("response").innerText = JSON.stringify(err.response, null, 2);
    });
  }
  </script>

</body></html>
```

Side notes:

1. Zoho's OAuth is not compatible with Pocketbase's built-in OIDC auth because they use the [ID Token](https://openid.net/specs/openid-connect-core-1_0.html#CodeIDToken) rather than a UserInfo API endpoint.
2. Zoho's ID token `claim["iss"]` vary from their regional domains, so [claims.VerifyIssuer()](https://github.com/pocketbase/pocketbase/blob/ad92992324c3fd0f09c36300890620a0224c6c06/tools/auth/apple.go#L145C13-L145C25) function won't be used.

### CODE REVIEW

**Comment by cxumol** [comment]
Let me explain in detail why I said "_their UserInfo endpoint is broken_", by elaborating two reproducible attempts.

#### Attempt 1

Based on [Zoho OIDC document](https://www.zoho.com/accounts/protocol/oauth/sign-in-using-zoho.html):
- Three endpoints `authorization_endpoint`(AuthURL), `token_endpoint`(TokenURL), and `userinfo_endpoint` (UserAPIURL) can be found from [Zoho OIDC metadata URL](https://accounts.zoho.com/.well-known/openid-configuration)
- Scopes can be `email`, `profile`, or `openid`

Settings on Admin UI
```yml
AuthURL: https://accounts.zoho.com/oauth/v2/auth
TokenURL: https://accounts.zoho.com/oauth/v2/token
UserAPIURL: https://accounts.zoho.com/oauth/v2/userinfo
```

In HTML Client
```js
pb.collection('users').authWithOAuth2({ provider: 'oidc', scopes: ["profile"] })
```

PocketBase console log
```
ERROR POST /api/collections/users/auth-with-oauth2
└─ Something went wrong while processing your request.
   └─ failed to fetch OAuth2 user profile via https://accounts.zoho.com/oauth/v2/userinfo (400):
...
                <html>
                <head>
                <title>Zoho Accounts</title>
                <meta name="viewport" content="width=device-width, initial-scale=1.0">
                <link href="https://static.zohocdn.com/iam/v2/components/css/zohoPuvi.c86bbb480e4a4fbc379fd8e7298bbde5.css" rel="stylesheet" type="text/css" >
                </head>
                <style>
                body {
                        width: 100%;
...
<div class="heading">An error occurred</div>
<div class="discrption">An error occurred while processing your request.<br> Please try again later.</div>
...
</html>
```

This is not a temporary error because I have been able to reproduce it over the past year. In addition, I also tried different value of scopes and permutations of them, and obtained the same results.

#### Attempt 2

By reading [a Stack Overflow answer](https://stackoverflow.com/a/54449818), I also tried "https://accounts.zoho.com/oauth/user/info" with "aaaserver.profile.READ" scope.

Settings on Admin UI
```yml
AuthURL: https://accounts.zoho.com/oauth/v2/auth
TokenURL: https://accounts.zoho.com/oauth/v2/token
UserAPIURL: https://accounts.zoho.com/oauth/user/info
```

In HTML Client
```js
pb.collection('users').authWithOAuth2({ provider: 'oidc', scopes: ["aaaserver.profile.READ"] })
```

PocketBase console log
```
ERROR POST /api/collections/users/auth-with-oauth2
└─ Failed to authenticate.
   └─ Missing required ExternalAuth fields.
```

The error message "Missing required ExternalAuth fields" seems some missing necessary parameters, but based on my understanding of [this thread](https://github.com/pocketbase/pocketbase/discussions/3749#discussioncomment-7564432), here the error could be because the `/oauth/user/info` UserInfo response does not comply with the [OIDC definition of a UserInfo Response](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims). And an example response from this endpoint can also be found from the same [Stack Overflow answer](https://stackoverflow.com/a/54449818). BTW, I'm using Stack Overflow as the reference because this API is [undocumented](https://help.zoho.com/portal/en/community/topic/oauth2-api-better-document-oauth-user-info). After all, this endpoint was never claimed to be an OIDC one.

In summary, the documented UserInfo endpoint always returns an error, and the undocumented alternative is working, but it does not comply with OIDC specifications. So I said "_their UserInfo endpoint is broken_".

Given that in the current PocketBase OIDC implementation the "User API URL" is a mandatory field, the current generic OIDC provider doesn't work with Zoho Auth in all of my attempts.

I'm so happy to look forward to a more customizable PocketBase in the upcoming versions, and thanks for all the good work you've been doing.

**Comment by cxumol** [comment]
I tested Zoho provider with the latest commit of this PR https://github.com/pocketbase/pocketbase/pull/5639/commits/f8c7ecf0f855b5358a498e3c050ff634f0df9cdf using v0.23.0 OAuth2 provider configuration on localhost, and it worked as expected.

**Comment by ganigeorgiev** [comment]
I've updated in the `develop` branch the existing OIDC implementation to support reading the user info from the `id_token` payload.

I've tested it also with Zoho and it works correctly.

Note that in the OIDC settings in the dashboard you'll find 2 new field options: "JWKS verification URL" and "Issuers". Both are optional and not needed for the proper OAuth2 flow and are there just for extra correctness.
